File: src/SIL.XForge.Scripture/Models/ParatextNoteThread.cs
using System.Collections.Generic;
using SIL.XForge.Models;

namespace SIL.XForge.Scripture.Models
{
    public class ParatextNoteThread : ProjectData
    {
        public string DataId { get; set; }
        public VerseRefData VerseRef { get; set; }
        public List<Note> Notes { get; set; } = new List<Note>();
        public string SelectedText { get; set; }
        public string ContextBefore { get; set; }
        public string ContextAfter { get; set; }
        // May be able to remove this altogether
        public int StartPosition { get; set; }
        public SegmentSelection CurrentContextSelection { get; set; }
        public string ParatextUser { get; set; }
        public string TagIcon { get; set; }
    }
}

File: src/RealtimeServer/scriptureforge/models/paratext-note-thread.ts
import { ProjectData, PROJECT_DATA_INDEX_PATHS} from "../../common/models/project-data";
import { ParatextNote } from "./paratext-note";
import { VerseRefData } from "./verse-ref-data";

export const PARATEXT_NOTE_THREAD_COLLECTION = "note_threads"
export const PARATEXT_NOTE_THREAD_INDEX_PATHS = PROJECT_DATA_INDEX_PATHS

export interface ParatextNoteThread extends ProjectData {
  dataId: string;
  verseRef: VerseRefData;
  notes: ParatextNote[];
  selectedText: string;
}
